sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -1347,6 +1400,41 @@ def plot(
                 f" of such values. Currently, kind={self.kind!r}"
             )
 
+        # FIXME: remove in 1.3
+        if self.pdp_lim != "deprecated":
+            warnings.warn(
+                "The `pdp_lim` parameter is deprecated in version 1.1 and will be "
+                "removed in version 1.3. Provide `pdp_lim` to the `plot` method."
+                "instead.",
+                FutureWarning,
+            )
+            if pdp_lim is not None and self.pdp_lim != pdp_lim:
+                warnings.warn(
+                    "`pdp_lim` has been passed in both the constructor and the `plot` "
+                    "method. For backward compatibility, the parameter from the "
+                    "constructor will be used.",
+                    UserWarning,
+                )
+            pdp_lim = self.pdp_lim
+
+        if pdp_lim is None:
+            # get global min and max average predictions of PD grouped by plot type
+            pdp_lim = {}
+            for idx, pdp in enumerate(self.pd_results):
+                values = pdp["values"]
+                preds = pdp.average if kind[idx] == "average" else pdp.individual
+                if centered and kind[idx] in ("both", "individual"):
+                    preds_offset = preds[self.target_idx, :, 0, None]
+                else:
+                    preds_offset = 0.0
+                min_pd = (preds[self.target_idx] - preds_offset).min()
+                max_pd = (preds[self.target_idx] - preds_offset).max()
+                n_fx = len(values)
+                old_min_pd, old_max_pd = pdp_lim.get(n_fx, (min_pd, max_pd))
+                min_pd = min(min_pd, old_min_pd)
+                max_pd = max(max_pd, old_max_pd)
+                pdp_lim[n_fx] = (min_pd, max_pd)
```

Comment:
Should we center before computing `pdp_lim`?

Reply:
Currently the limits are just a bit wider than if they were computed after the centering, but never too narrow. Can we merge as is and do a follow up PR to better adjust the limits ? @glemaitre  @thomasjpfan  ?

Reply:
I would merge it now and make a small subsequent PR.

Reply:
I'm okay to merge first if @jeremiedbb and @jjerphan are okay with it.

I think that fixing `pdp_lim` to take into account of centering is a blocker for release tho. Imagine a case where the first point is `[0, -10]` and the last point is `[0, 10]`, the range is (-10, 10), which will cut off the plot after centering.